hack/consecutive-upgrades-test.sh

```diff
@@ -0,0 +1,283 @@
+sudo chown -R \${NEW_USER}: /home/\${NEW_USER}/.ssh
+sudo chmod 600 /home/\${NEW_USER}/.ssh/authorized_keys
+END
+
+CSV=$( ${CMD} get csv -o name -n ${HCO_NAMESPACE} | grep "kubevirt-hyperconverged-operator")
+
+Msg "operator conditions before upgrade"
+source ./hack/check_operator_condition.sh
+KUBECTL_BINARY=${CMD} INSTALLED_NAMESPACE=${HCO_NAMESPACE} printOperatorCondition "${INITIAL_VERSION}"
+
+### Create a VM ###
+Msg "Create a simple VM on the previous version cluster, before the upgrade"
+${CMD} get namespace | grep "^${VMS_NAMESPACE}" || ${CMD} create namespace ${VMS_NAMESPACE}
+${CMD} get secret -n ${VMS_NAMESPACE} | grep "^testvm-secret" || ${CMD} create secret -n ${VMS_NAMESPACE} generic testvm-secret --from-file=userdata=./hack/cloud-init.sh
+${CMD} apply -n ${VMS_NAMESPACE} -f ./hack/vm.yaml
+${CMD} get vm -n ${VMS_NAMESPACE} -o yaml testvm
+~/virtctl start testvm -n ${VMS_NAMESPACE}
+./hack/retry.sh 30 10 "${CMD} get vmi -n ${VMS_NAMESPACE} testvm -o jsonpath='{ .status.phase }' | grep 'Running'"
+${CMD} get vmi -n ${VMS_NAMESPACE} -o yaml testvm
+
+upgrade $INITIAL_VERSION $MID_VERSION $OO_MID_BUNDLE
+upgrade $MID_VERSION $TARGET_VERSION $OO_LAST_BUNDLE
+
+
+Msg "make sure that we don't have outdated VMs"
+
+INFRASTRUCTURETOPOLOGY=$(${CMD} get infrastructure.config.openshift.io cluster -o json | jq -j '.status.infrastructureTopology')
+UPDATE_METHODS=$(${CMD} get hco ${HCO_RESOURCE_NAME} -n ${HCO_NAMESPACE} -o jsonpath='{.spec .workloadUpdateStrategy .workloadUpdateMethods}')
+
+if [[ "${INFRASTRUCTURETOPOLOGY}" == "SingleReplica" ]]; then
+  echo "Skipping the check on SNO clusters"
+elif [[ "${UPDATE_METHODS}" == "" || "${UPDATE_METHODS}" == "[]" ]]; then
+  echo "Skipping while workloadUpdateMethods methods are empty "
+else
+  ./hack/retry.sh 10 30 "[[ \$(${CMD} get vmi -l kubevirt.io/outdatedLauncherImage -A --no-headers | wc -l) -eq 0 ]]" "${CMD} get vmi -l kubevirt.io/outdatedLauncherImage -A"
+  echo "All the running VMs got upgraded"
+fi
+
+./hack/retry.sh 5 30 "~/virtctl stop testvm -n ${VMS_NAMESPACE}"
+${CMD} delete vm -n ${VMS_NAMESPACE} testvm
+
+KUBECTL_BINARY=${CMD} ./hack/test_quick_start.sh
+
+Msg "Check that OVS is deployed or not deployed according to deployOVS annotation in HCO CR."
+./hack/retry.sh 40 15 "CMD=${CMD} PREVIOUS_OVS_ANNOTATION=${PREVIOUS_OVS_ANNOTATION}\
+ PREVIOUS_OVS_STATE=${PREVIOUS_OVS_STATE} ./hack/check_upgrade_ovs.sh"
+
+Msg "Ensure that console plugin deployment and service has been renamed successfully"
+KUBECTL_BINARY=${CMD} INSTALLED_NAMESPACE=${HCO_NAMESPACE} ./hack/check_upgrade_console_plugin.sh
+
+Msg "check virtio-win image is in configmap"
+VIRTIOWIN_IMAGE_CSV=$(${CMD} get ${CSV} -n ${HCO_NAMESPACE} \
+  -o jsonpath='{.spec.install.spec.deployments[?(@.name=="hco-operator")].spec.template.spec.containers[0].env[?(@.name=="VIRTIOWIN_CONTAINER")].value}')
+VIRTIOWIN_IMAGE_CM=$(${CMD} get cm virtio-win -n ${HCO_NAMESPACE} -o jsonpath='{.data.virtio-win-image}')
```

Comment:
not sure about these checks. I would drop them to simplify the test, and because we're checking that in the regular upgrade tests. I think it's not the focus of this test.

Reply:
@dharmit - could you remove this one too?